crates/proc-macro-api/src/process.rs

```diff
@@ -71,7 +70,9 @@ impl ProcMacroServerProcess {
             Ok(v) => {
                 tracing::info!("Proc-macro server version: {v}");
                 srv.version = v;
-                if srv.version >= RUST_ANALYZER_SPAN_SUPPORT {
+                if srv.version >= HASHED_AST_ID {
+                    // We don't enable spans on versions prior to `HASHED_AST_ID`, because their ast id layout
+                    // is different.
```

Comment:
Given we are bumping the protocol in an incompatible-ish way again anyways, we might wanna consider informign the server about the span that is considered fake. That way we have leeway in changing its representation later on if need be? (i always regretted not making that part of the protocol but instead just having it hardcoded ...)

That is `FIXUP_ERASED_FILE_AST_ID_MARKER`